report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/InstantUtil.java

```diff
@@ -0,0 +1,50 @@
+package no.sikt.nva.data.report.api.fetch.model;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import nva.commons.apigateway.exceptions.BadRequestException;
+
+public final class InstantUtil {
+
+    private static final String TIME_SEPARATOR = "T";
+
+    private static final LocalTime END_OF_DAY = LocalTime.of(23, 59, 59);
+    public static final String DATE_FORMAT_ERROR = "Supply a valid '%s' value in format 'YYYY-MM-DD' or "
+                                                   + "'YYYY-MM-DDThh:mm:ssZ'";
+    public static final String BEFORE_TEXT = "before";
+    public static final String AFTER_TEXT = "after";
+
+    private InstantUtil() {
+        // NO-OP
+    }
+
+    public static Instant before(String date) throws BadRequestException {
```

Comment:
Confused by the method names `before` and `after`, you are parsing a String to an Instant at either start of day or end of day? Rename methods to `parseAtStartOfDay` and `parseAtEndOfDay` or something?

Reply:
Maybe confused because I associate the method names with 
[isAfter(OffsetDateTime other) and  isBefore(OffsetDateTime other)](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html)

Reply:
Agreed. We need to work on the terminology generally in the API